runtime/dart_isolate.cc

```diff
@@ -391,33 +393,49 @@ bool DartIsolate::MarkIsolateRunnable() {
 }
 
 FML_WARN_UNUSED_RESULT
-bool DartIsolate::Run(const std::string& entrypoint_name) {
-  TRACE_EVENT0("flutter", "DartIsolate::Run");
-  if (phase_ != Phase::Ready) {
+static bool InvokeMainEntrypoint(Dart_Handle user_entrypoint_function) {
+  if (tonic::LogIfError(user_entrypoint_function)) {
+    FML_LOG(ERROR) << "Could not resolve main entrypoint function.";
     return false;
   }
 
-  tonic::DartState::Scope scope(this);
+  auto run_main_zoned_function =
+      Dart_GetField(Dart_LookupLibrary(tonic::ToDart("dart:ui")),
+                    tonic::ToDart("_runMainZoned"));
 
-  Dart_Handle entrypoint =
-      Dart_GetField(Dart_RootLibrary(), tonic::ToDart(entrypoint_name.c_str()));
-  if (tonic::LogIfError(entrypoint)) {
+  auto start_main_isolate_function =
+      Dart_GetField(Dart_LookupLibrary(tonic::ToDart("dart:isolate")),
+                    tonic::ToDart("_startMainIsolate"));
+
+  if (tonic::LogIfError(run_main_zoned_function) ||
+      tonic::LogIfError(start_main_isolate_function)) {
+    FML_LOG(ERROR) << "Could not resolve main entrypoint trampolines.";
     return false;
   }
 
```

Comment:
You could invoke runMainZoned directly instead of closurizing first. Instead of

Dart_Invoke(Dart_GetField(dart:ui, "_runMainZoned"), a, b)

do

Dart_Invoke(dart:ui, "_runMainZoned", a, b)

Reply:
I was attempting to use the tonic wrappers for this.

Reply:
This caused breakage: note that `@pragma('vm:entry-point')` does not permit closurization of the static methods - it only permits invocation. Thus AOT compiler retained `_runMainZoned` but not `get:_runMainZoned` which is necessary for closurization. This subsequently crashes in AOT mode because runtime can't create `get:_runMainZoned` dynamically. 

I am going to revert this PR for now to unbreak the engine. 

I have also bumped priority of dart-lang/sdk#35369 - which should catch this sort of issues in the JIT mode as well, not just in AOT mode.

/cc @sjindel-google